tls/crypto.c

```diff
@@ -442,6 +457,25 @@ ttls_crypto_modinit(void)
 		}
 	}
 
+	/*
+	 * Below, so that when the "shaN" request is loaded, the driver with the
+	 * highest priority is loaded, even if some one with a lower priority is
+	 * already loaded, you must try to load all cripto kernel modules.
```

Comment:
I'd change that to something like:
"Each hash algorithm may have multiple implementations, optimized for different instruction set extensions (such as AVX, AVX2, and so on). As we want the fastest available, we try to load all the implementations. Crypto API will then choose one with the highest priority."